src/providers/VoiceFocusProvider/index.tsx

```diff
@@ -165,21 +175,25 @@ const VoiceFocusProvider: React.FC<Props> = ({ spec, options, children }) => {
   async function initVoiceFocus(
     vfSpec: VoiceFocusSpec | undefined,
     options: VoiceFocusDeviceOptions | undefined,
+    createMeetingResponse: any | undefined,
     canceled: () => boolean
   ) {
     // Throw away the old one and reinitialize.
     setVoiceFocusTransformer(null);
     setVoiceFocusDevice(null);
-    createVoiceFocusDeviceTransformer(vfSpec, options, canceled);
+    createVoiceFocusDeviceTransformer(vfSpec, options, createMeetingResponse, canceled);
   }
 
   useEffect(() => {
     let canceled = false;
-    initVoiceFocus(vfSpec, options, () => canceled);
+    if (createMeetingResponse && !initialized) {
+      initVoiceFocus(vfSpec, options, createMeetingResponse, () => canceled);
+      setInitialized(true);
+    }
```

Comment:
But `initVoiceFocus` can fail, and is asynchronous. This logic doesn't seem right. Try it: turn your wifi off after loading the page but before mounting the component. You'll end up with `setInitialized(true)` being called, but the transformer is not initialized.

Reply:
`initVoiceFocus` won't be called until `createMeetingResponse` is not null. I tried to replace boolean `initialized` with checking existence of transformer, due to `initVoiceFocus` is async, so I see this useEffect is called multiple times before transformer is actually created, which results in different complexity models getting downloaded multiple times.

Reply:
figured out the root cause here. reason why this useeffect is called multiple times is that vfSpec is not deep compared. Adding `JSON.stringify(prev) === JSON.stringify(next)` solved the problem. Will removed `setInitialized(true)`.